### PR TITLE
Do not test `GatewayClient` on integration

### LINKED ISSUE #974: Replace integration tests in `gateway_test.py`
Some tests in `gateway_test.py` run on integration network - they have a comment

```
# Temporary test to be replaced after devnet supports new starknet
```

### PR DESCRIPTION
**Pull request by war-in**
Closes #974 

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/tests/e2e/client/fixtures/transactions.py [hunk 1]
```diff
@@ -53,3 +60,73 @@ def block_with_deploy_account_number(
...
+    await deploy_result.wait_for_acceptance()
+
+    contract = deploy_result.deployed_contract
+
+    resp = await contract.functions["replace_implementation"].invoke(
+        new_class=map_class_hash, max_fee=MAX_FEE
+    )
+    await resp.wait_for_acceptance()
+
+    return resp.block_number, contract.address, map_class_hash
```
Comment (drknzz) [inline starknet_py/tests/e2e/client/fixtures/transactions.py:132]: Are we testing anywhere the class replacement? Maybe let's add a test for the new `replace_class.cairo` to make sure it's working properly

Comment (war-in) [reply starknet_py/tests/e2e/client/fixtures/transactions.py]: I don't think it's necessary. The test is working properly

File: starknet_py/tests/e2e/client/fixtures/transactions.py [hunk 2]
```diff
@@ -1,20 +1,27 @@
+# pylint: disable=redefined-outer-name
```
[the remark this answers is not in the document]
Comment (war-in) [reply starknet_py/tests/e2e/client/fixtures/transactions.py:1]: to the pytest_asyncio extension
image

[the remark this answers is not in the document]
Comment (drknzz) [reply starknet_py/tests/e2e/client/fixtures/transactions.py:1]: Could we add a comment / bring it closer to the issue?

File: starknet_py/tests/e2e/client/fixtures/transactions.py (hunk 1 above)
Comment (drknzz) [reply starknet_py/tests/e2e/client/fixtures/transactions.py]: I thought about some test that would use the contract after replacing the class

Comment (war-in) [reply starknet_py/tests/e2e/client/fixtures/transactions.py]: But I'm not sure if we should. That would not test the starknet.py feature, just cairo contract

File: starknet_py/tests/e2e/client/fixtures/transactions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (war-in) [reply starknet_py/tests/e2e/client/fixtures/transactions.py:1]: We could but it is already present in many files with fixtures
image

**Review by drknzz — APPROVED** [review]
👍

File: starknet_py/tests/e2e/client/fixtures/transactions.py (hunk 1 above)
Comment (war-in) [reply starknet_py/tests/e2e/client/fixtures/transactions.py]: Checked it locally and it works 🎉

**Pull request merged**